proposals/cli_schema.md

```diff
@@ -0,0 +1,246 @@
+
+## Proposal
+
+This provides a high level overview of the feature.
+
+A new schema is being proposed for the `test` command which will provide the necessary structure to account for all major current capabilities of Kyverno and close some issues related to problems. There are some high-level functional requirements that must be met.
+
+1. The structure must be completely validated.
+2. It must allow backwards compatibility with existing test manifests out in the wild.
+   1. A test manifest which supplies the `apiVersion` and `kind` as specified in the proposed schema below is not subject to this backwards compatibility; only those in which these fields are absent should be subject to the existing logic.
+3. All file names must be checked for existence and validity.
+4. Objects in the `results[]` section should be validated against the rule type to determine if the necessary mandatory fields are present.
+5. Policies which are of type `Policy` must specify Namespace.
+6. There must not be duplicate names in resources.
+7. Deprecated fields must be handled with warnings to users.
+8. Breaking changes should cause the apiVersion to be incremented.
+
+```yaml
+apiVersion: cli.kyverno.io/v1beta1
+kind: KyvernoTest
+metadata:
+  name: test-check-nvidia-gpus
+  labels:
+    foolabel: foovalue
+  annotations:
+    fookey: foovalue
+spec:
+  policies:
+    # support fully-qualified file paths (https://github.com/kyverno/kyverno/issues/2315)
+    # wherever files are the value. Should be relative and absolute.
+  - check-nvidia-gpu.yaml
+  - relative/path/to/next/policy.yaml
+  - /fully/qualified/path/to/next/policy.yaml
+  resources:
+  - resource.yaml
+  - relative/path/to/next/resource.yaml
+  - /fully/qualified/path/to/next/resource.yaml
+  results:
+  # results[] vary depending on the rule type being tested.
+  # validate
+  - policy: policyname01 # required
+    rule: validaterulename # required
+    # resources must support multiple resources, all of which
+    # should be subject to the rest of the field values 
+    # in this result. (https://github.com/kyverno/kyverno/issues/2857)
+    resources: # required
+    - resource01
+    - resource02
+    - resource03
+    # namespace should set the namespace name for the resource(s), if not present,
+    # but otherwise should override the Namespace name if present. If the policy
+    # being tested is a `Policy` (Namespaced), and if the namespace is not defined
+    # in the resource AND not defined here, it should assume the value of `default`
+    # to be consistent with how Kubernetes treats resources (i.e., creating a resource
+    # with no metadata.namespace field implies the default namespace) (https://github.com/kyverno/kyverno/issues/4059)
+    # (https://github.com/kyverno/kyverno/issues/4087)
+    namespace: foonamespace # optional
+    kind: Kind # required
```

Comment:
Is this the resources' kind? Do we want to extend it to support group/version/kind?

Reply:
Yes, good point. It should support format of Kind or GVK.

Reply:
Captured in https://github.com/kyverno/KDP/pull/21/commits/6f7d504b76e87f91a38d9aba1d6fd6ef26015b0d